File: Week2/Homework/hw2-2/hw2-2.js
student_ids = db.grades.distinct('student_id');
for (var i=0; i<student_ids.length; i++) {
  homework = db.grades.find({student_id: student_ids[i],type:'homework'}).sort({score:1}).limit(1);
  db.grades.remove({_id: homework[0]._id});
}

File: Week2/Homework/hw2-2/hw2-2.pl
#!/usr/bin/perl
use MongoX (host => 'mongodb://127.0.0.1');
use Tie::IxHash;

with_context {
    my $student_id_last = -1;
    my $grades = context_collection
                    ->find({type => 'homework'})
                    ->sort(Tie::IxHash->new(student_id => 1, score => 1));
    while(my $grade = $grades->next()) {
        if($student_id_last != $grade->{student_id}){
            $student_id_last = $grade->{student_id};
            context_collection->remove($grade);
        }
    }
} db => 'students', collection => 'grades';
